apps/web/src/app/_components/header.tsx

```diff
@@ -21,6 +23,31 @@ interface HeaderProps {
  */
 export default function Header({ session, auth }: HeaderProps) {
   const pathname = usePathname();
+  const [showButton, setShowButton] = useState(true);
+
+  const profile = api.profile.getCurrentUser.useQuery();
+  const [profileId, setProfileId] = useState<string | undefined>(
+    profile.isSuccess ? profile.data?.id : undefined,
+  );
```

Comment:
We don't need state management for this since React Query will handle the changes in data by triggering a re-render when the query data changes.

```suggestion
  const profileId = profile.data?.id;
```